### ISSUE TITLE
browser signs me out of google services everytime time i close the browser

### ISSUE THREAD

**Issue by Lukrativehd** [Alex313031/thorium#715]
**System Details**
 - OS: Windows 11
 - Thorium Version 124.0.6367.218 (Official Build) (64-bit)

**Problem**
Each time i close the browser im forced to sign back into my gmail and youtube and anything related to google, which is a real PITA as i have 2fa on everything if i havent got my phone to hand i cant access my emails etc. I only have one drive, C drive with no partitions, I dont dont see any settings to help me identify what the issue may be.

**Comment by gz83** [comment]
Please upload the flags that your browser is currently using

**Comment by Lukrativehd** [comment]
[Thorium Flags.pdf](https://github.com/user-attachments/files/15846373/Thorium.Flags.pdf)

**Comment by Lukrativehd** [comment]
whats the best way to do it, its like 72 pages long and i cannot upload html files to github? 

@gz83

**Comment by gz83** [comment]
What I mean is the experimental flags that the browser is currently using, you can see them by going to the chrome://versions page, take a screenshot of the chrome://versions page and upload the screenshot here.

@Lukrativehd

**Comment by Lukrativehd** [comment]
![Screenshot_13](https://github.com/Alex313031/thorium/assets/5403740/2f5b9006-2872-48dd-bc2e-26f84e8e6ed5)

**Comment by Lukrativehd** [comment]
the page doesnt exist ? 
 @gz83

**Comment by Lukrativehd** [comment]
![Screenshot_14](https://github.com/Alex313031/thorium/assets/5403740/df665967-37ab-4bbd-a139-e6f5dbd5f89b)
![Screenshot_15](https://github.com/Alex313031/thorium/assets/5403740/b9379ca9-97b0-4b9f-8dc4-2dccd86d3137)


or do you mean these?

**Comment by Lukrativehd** [comment]
![Screenshot_16](https://github.com/Alex313031/thorium/assets/5403740/6e0cfcf0-2b66-4af3-870d-d995a23b63b0)

however chrome://version works without the 's' and this is where it brings me to

**Comment by gz83** [comment]
Sorry, I typed an extra (s) in the previous comment. From the screenshots you uploaded recently, it seems that the problem is not caused by flags. How did you install the browser? Was the user data migrated from another browser?

**Comment by Lukrativehd** [comment]
i installed from the github downloading from this link off the site : https://github.com/Alex313031/Thorium-Win/releases

And once installed i logged into my google account and chose to sync settings such as bookmarks and passwords etc, this is a brand new fresh install of windows and the first browser to be installed

@gz83

**Comment by starkwiz** [comment]
Don't want to hijack this thread, but I am facing exactly this issue and I upgraded from 119.0.6045.214 to 124.0.6367.218
I am using it in portable format, and I downloaded the AVX2 build.
I think whatever helps resolve this issue might help me too...

**Comment by Lukrativehd** [comment]
any updates for solutions as this issue is really starting to p*** me off im ngl having to find my phone to auth my emails and youtube everytime i close and reopen the browser is driving me nuts 

@gz83

**Comment by gz83** [comment]
Have you cleared your browser data (such as history, cookies, etc.) and logged into your Google account?

@Lukrativehd

**Comment by starkwiz** [comment]
I resolved this by updating **THORIUM.BAT** to not use **"--disable-encryption"**. Found this fix in another issue mentioned in Thorium github. I don't know if removing this parameter affects portability across machines.

My batch file looks like this now... indeed the line starting with "rem" is comment in batch file.
```
START "" "%cd%\BIN\thorium.exe" --user-data-dir="%~dp0%\USER_DATA" --allow-outdated-plugins --disable-logging --disable-breakpad --disable-machine-id
rem START "" "%cd%\BIN\thorium.exe" --user-data-dir="%~dp0%\USER_DATA" --allow-outdated-plugins --disable-logging --disable-breakpad --disable-encryption --disable-machine-id
```

**Comment by Lukrativehd** [comment]
this worked @starkwiz  i created the batch file ran it and it asked me to sign in again at the start and once i'd signed in i restarted and closed thorium a few times and every time i was still signed in! so for me moment it seems to have worked thank you so much <3

**Comment by kenniky** [comment]
Hi, I'm having this issue too. Where did you find/create the thorium.bat file? @starkwiz @Lukrativehd

**Comment by littleblack111** [comment]
same. i hav this issue too

**Comment by littleblack111** [comment]
but restarting the whole browser fixes it(i hav 2 profiles/windows, and closing both of them and restarting fixes it, pretty annoying still...)

**Comment by starkwiz** [comment]
> I resolved this by updating **THORIUM.BAT** to not use **"--disable-encryption"** parameter. Found this fix in another issue mentioned here in Thorium github. I don't know if removing this parameter affects portability across machines.
> 
> My batch file looks like this now... indeed the line starting with **"rem"** is comment in batch file.
> 
> ```
> START "" "%cd%\BIN\thorium.exe" --user-data-dir="%~dp0%\USER_DATA" --allow-outdated-plugins --disable-logging --disable-breakpad --disable-machine-id
> rem START "" "%cd%\BIN\thorium.exe" --user-data-dir="%~dp0%\USER_DATA" --allow-outdated-plugins --disable-logging --disable-breakpad --disable-encryption --disable-machine-id
> ```

I've realised that this workaround actually affects portability.
The real issue is that the parameters disable-encryption and disable-machine-id which are meant to enable portability across computers, don't work with old browser profile.
So, if you are facing this issue post-upgrade you probably have an old profile.

I don't think we have to re-create the whole USER_DATA folder but simply renaming the "Local State" file (not folder) in USER_DATA folder to "Local State.bak" should cause it to be recreated on next thorium start-up.
If you have setup multiple profiles there are chances that you might lose access to those, so make sure to take backup of this folder prior to making changes.

### TL;DR:  Simply renaming "Local State" file to "Local State.bak" in USER_DATA folder, should fix this issue. $\color{red}{\text{You don't have to modify the batch file.}}$
